src/cli/cmd_clean.py

```diff
@@ -0,0 +1,150 @@
+# pylint: disable=duplicate-code
+"""Module for the cleaning command."""
+import logging
+import sys
+from datetime import datetime
+from pathlib import Path
+from typing import Annotated, Optional
+
+import typer
+
+from src.blueprints.datafile import RawDatafile
+from src.cli.common import (
+    LogFileOption,
+    ProfileNameOption,
+    ProfileVersionOption,
+    SourceDataPathArg,
+    StorageBoxOption,
+    get_config,
+)
+from src.config.config import ConfigFromEnv
+from src.inspector.inspector import Inspector
+from src.profiles.profile_register import load_profile
+from src.utils import log_utils
+from src.utils.filesystem.ctime import max_ctime
+from src.utils.timing import Timer
+
+logger = logging.getLogger(__name__)
+
+
+def _check_verified_status(config: ConfigFromEnv, datafiles: list[RawDatafile]) -> None:
+    logger.info("Retrieving status...")
+    # Check ingestion status for each file.
+    timer = Timer(start=True)
+    insepctor = Inspector(config)
+    results = [insepctor.is_datafile_verified(raw_df) for raw_df in datafiles]
+    logger.info("Retrieved datafile status in %f seconds.", timer.stop())
+    # Get unverified files by the index of the verification results list.
+    unverified_dfs = [
+        datafiles[ind] for ind, is_verified in enumerate(results) if not is_verified
+    ]
+
+    if len(unverified_dfs) > 0:
+        logger.info("Datafiles pending ingestion or verification:")
+        for datafile in unverified_dfs:
+            logger.info(datafile.directory / datafile.filename)
+        logger.error(
+            "Could not proceed with deleting this data root. Ingestion is not complete."
+        )
+        sys.exit(1)
+    else:
+        logger.info("Ingestion for this data root is complete.")
+
+
+def _check_file_age(df_paths: list[Path], min_file_age: int) -> None:
+    logger.info("Checking file age...")
+    newest_ctime = max_ctime(df_paths)
+    days_since_ctime = (datetime.now() - datetime.fromtimestamp(newest_ctime)).days
+    if not days_since_ctime >= min_file_age:
```

Comment:
Do we want this to exit if a single file is too young? As with the previous function is it perhaps better to pass a filtered list of files that can be deleted and continue the script?

Reply:
I've refactored the code, so it is easy to implement a partial delete feature/flag. I have found that there is a larger issue with partially deleting files. The cleaning command reuses the profile extraction step. Profiles perform a checksum during extraction, and it raises an exception when a file is not found. Arguably, when we discover a specified datafile is missing, failing fast is useful when the extraction is done as part of ingestion, as it could flag an error in the data source.